Sources/SwiftProtobuf/BinaryEncoder.swift

```diff
@@ -125,11 +127,19 @@ internal struct BinaryEncoder {
 
     // Write a string field, including the leading index/tag value.
     mutating func putStringValue(value: String) {
-        let count = value.utf8.count
-        putVarInt(value: count)
-        for b in value.utf8 {
-            pointer.storeBytes(of: b, as: UInt8.self)
-            pointer = pointer.advanced(by: 1)
+        // If the String does not support an internal representation in a form
+        // of contiguous storage, body is not called and nil is returned.
+        let isAvailable = value.utf8.withContiguousStorageIfAvailable { (body: UnsafeBufferPointer<UInt8>) -> Int in
+            self.putVarInt(value: body.count)
```

Comment:
style wise, we don't do `self.`, mixing the style within a single function also really sticks out.

Reply:
Thank you for the feedback this should now be fixed in 11531df7a63cb0036775610ba14ff593732c8d7a.